### PR TITLE
Show revision in footer; clean up path and site setup

### PR DESCRIPTION
**Pull request by drkitty**
**Major changes**:
- **`wsgi.py` has been moved to the root of the repo.**
- You can use a virtualenv even via WSGI. `wsgi.py` automatically takes care of the details.
- To import Cyder or one of its libraries from a script, do `import activate; activate.activate()` instead of `import manage`.

Minor changes:
- The current revision is now displayed in the footer.
- If you use a virtualenv, you no longer need to activate it before running `manage.py`.
- Some incorrectly duplicated libraries were removed, so we no longer rely on a specific ordering of `sys.path`.
- To form an absolute path from a relative path from the root of the repo, use `cy_path` from the `activate` module.

### CODE REVIEW

File: bin/lib/recreate_database.py [hunk 1]
```diff
@@ -14,11 +16,11 @@ def main():
     print '****************************************************'
     print
     while True:
-        response = raw_input('Proceed [y/n]? ')
-        if response.lower() == 'y':
+        response = raw_input('Proceed [yes/no]? ')
```
Comment (murrown) [inline bin/lib/recreate_database.py]: This is nitpicking but I think you should leave this as `y/n`, or at least accept `y` as an alias for `yes`. My reasoning for this is that the `yes` Unix command outputs `y` by default. One of my own scripts has a line that reads `yes | bash bin/recreate_database.sh`, haha. Plus, a lot of users are conditioned by other applications to just type `y`.

If you're really concerned that someone will drop the database unintentionally then I'll concede but I thought I'd make my case anyway.

File: requirements/dev.txt [hunk 2]
```diff
@@ -15,5 +15,5 @@ Sphinx==1.0.7
 # Testing
 nose==1.0.0
 mock==0.7.2
--e git://github.com/jbalogh/django-nose.git#egg=django_nose
--e git://github.com/jbalogh/test-utils.git#egg=test-utils
+git+git://github.com/jbalogh/django-nose.git#egg=django_nose
+git+git://github.com/jbalogh/test-utils.git#egg=test-utils
```
Comment (murrown) [inline requirements/dev.txt]: This was changed because the -e flag was removed in newer versions of pip?

**Comment by murrown** [comment]
There's a bit of outdated documentation that references tastypie in `cyder/cydns/models.py`

File: requirements/prod.txt [hunk 3]
```diff
@@ -38,37 +38,18 @@ versiontools==1.9.1
...
-
-# Various tidbits
--e git://github.com/jsocol/commonware.git#egg=commonware
--e git://github.com/mozilla/nuggets.git#egg=nuggets
-
-# Security
--e git://github.com/fwenzel/django-sha2.git#egg=django-sha2
--e git://github.com/jsocol/bleach.git#egg=bleach
--e git://github.com/mozilla/django-session-csrf.git#egg=django-session-csrf
+git+git://github.com/mozilla/happyforms.git#egg=happyforms
```
Comment (murrown) [inline requirements/prod.txt]: Are we even really using happyforms? It doesn't look like Cyder ignores leading whitespace on my end.

File: requirements/dev.txt (hunk 2 above)
Comment (drkitty) [inline requirements/dev.txt]: I'm pretty sure the flag is still supported. It installs the packages in "develop" mode, and while I don't completely understand its effects, it's certainly not necessary.

File: requirements/prod.txt (hunk 3 above)
Comment (drkitty) [inline requirements/prod.txt]: No idea. I doubt it.

File: bin/lib/recreate_database.py (hunk 1 above)
Comment (drkitty) [inline bin/lib/recreate_database.py]: The fact that users are used to typing "y" is the reason I chose to make it longer. Dropping a database accidentally is about the worst thing I can think of happening.

**Comment by drkitty** [comment]
Re: the stuff in `cyder/cydns/models.py`, I'll take a look at that.

File: bin/lib/recreate_database.py (hunk 1 above)
Comment (murrown) [inline bin/lib/recreate_database.py]: I agree, but I don't think making them type out the full word is going to offer any more security if they're already determined enough to run a script called `recreate_database.py` and respond to the warning in the affirmative.

**Comment by murrown** [comment]
I'm ready to merge this as-is though, the stuff I pointed out was what I considered noteworthy but none of it is deal-breaking. When you think this pull request is ready, I say go ahead and merge it.

**Comment by drkitty** [comment]
Ready to go.

**Pull request merged**